src/components/MovieDetails.tsx

```diff
@@ -1,9 +1,108 @@
-const MovieDetails = () => {
-  return (
-    <div>
-      <p>Movie details</p>
+import { useState, useEffect } from 'react';
+
+import api from 'src/service/api';
+
+import ReadMore from './ReadMore';
+
+import '../styles/MovieDetails.scss';
+
+type MovieDetailsProps = {
+  match: { params: { id: string } }
+}
+
+const MovieDetails = ({ match }: MovieDetailsProps) => {
+
+  const [movieId, setMovieId] = useState<string>(' ');
+  const [movieData, setMovieData] = useState({
+    poster_path: '',
+    original_title: '',
+    title: '',
+    tagline: '',
+    status: '',
+    release_date: '',
+    budget: '',
+    popularity: '',
+    vote_average: '',
+    runtime: '',
+    genres: [{ id: null, name: ' ' }],
+    production_countries: [{ name: '' }],
+    overview: '',
+  })
+
+  const {
+    poster_path,
+    original_title,
+    title,
+    tagline,
+    status,
+    release_date,
+    budget,
+    popularity,
+    vote_average,
+    runtime,
+    genres,
+    production_countries,
+    overview
+  } = movieData;
+
+  const poster = api.changeImgLinks(poster_path, 'w500');
```

Comment:
'w500' - such values looks like Enum and should be added as Enum

Reply:
done